UnityProject/Assets/Scripts/UI/Systems/AdminTools/PlayerManagePage.cs

```diff
@@ -213,18 +213,15 @@ private void SendTeleportAllPlayersToPlayer()
 		public void GiveItemToPlayerButton()
 		{
 			adminTools.giveItemPage.selectedPlayer = null;
-			var players = FindObjectsOfType<PlayerScript>(); //since this is client sided it's fiinnnneee
-			foreach (var possiblePlayer in players)
-			{
-				if(possiblePlayer.PlayerInfo.Username != PlayerEntry.PlayerData.accountName) continue;
-				adminTools.giveItemPage.selectedPlayer = possiblePlayer.gameObject;
-			}
+
+			adminTools.giveItemPage.selectedPlayer = PlayerList.Instance.GetPlayerByID(PlayerEntry.PlayerData.uid).GameObject;
```

Comment:
The removed comment says this is clientside, so i don't think this will work as Playerlist stuff is serverside